treeherder/log_parser/utils.py

```diff
@@ -1,66 +1,31 @@
-import logging
-
 import simplejson as json
-from requests.exceptions import HTTPError
-
-from treeherder.etl.artifact import (serialize_artifact_json_blobs,
-                                     store_job_artifacts)
-from treeherder.log_parser.artifactbuildercollection import (ArtifactBuilderCollection,
-                                                             LogSizeException)
-from treeherder.model.models import JobLog
-
-logger = logging.getLogger(__name__)
-
-
-def extract_text_log_artifacts(job_log):
-    """Generate a set of artifacts by parsing from the raw text log."""
-
-    # parse a log given its url
-    artifact_bc = ArtifactBuilderCollection(job_log.url)
-    artifact_bc.parse()
-
-    artifact_list = []
-    for name, artifact in artifact_bc.artifacts.items():
-        artifact_list.append({
-            "job_guid": job_log.job.guid,
-            "name": name,
-            "type": 'json',
-            "blob": json.dumps(artifact)
-        })
+from jsonschema import (ValidationError,
+                        validate)
 
-    return artifact_list
+with open('schemas/performance-artifact.json') as f:
+    PERFHERDER_SCHEMA = json.load(f)
 
 
-def post_log_artifacts(job_log):
-    """Post a list of artifacts to a job."""
-    logger.debug("Downloading/parsing log for log %s", job_log.id)
+def validate_perf_data(performance_data: dict):
+    validate(performance_data, PERFHERDER_SCHEMA)
+    expected_range = (45, PERFHERDER_SCHEMA["definitions"]["suite_schema"]["properties"]["extraOptions"]["items"]["maxLength"])
+    # allow only one extraOption longer than 45
+    if len(_long_elements(expected_range, _extra_options(performance_data))) > 1:
```

Comment:
This isn't doing what it should.
That is, if we have at least 2 suites & each one has a single extra options field between 45 and 100, this will raise `ValidationError` & we don't want that.

Reply:
Indeed, there was a problem. I solved it and tested this use case in the test_validate_perf_schema_no_exception test case, which tests the above situation with 3 suites.